propagation.go

```diff
@@ -1,246 +1,69 @@
 )
 
-type splitTextPropagator struct {
+type accessorPropagator struct {
 	tracer *tracerImpl
 }
-type splitBinaryPropagator struct {
-	tracer *tracerImpl
-}
-type goHTTPPropagator struct {
-	*splitBinaryPropagator
-}
-
-const (
-	fieldNameTraceID = "traceid"
-	fieldNameSpanID  = "spanid"
-	fieldNameSampled = "sampled"
-)
-
-func (p *splitTextPropagator) InjectSpan(
-	sp opentracing.Span,
-	carrier interface{},
-) error {
-	sc := sp.(*spanImpl)
-	splitTextCarrier, ok := carrier.(*opentracing.SplitTextCarrier)
-	if !ok {
-		return opentracing.ErrInvalidCarrier
-	}
-	if splitTextCarrier.TracerState == nil {
-		splitTextCarrier.TracerState = make(map[string]string, 3 /* see below */)
-	}
-	splitTextCarrier.TracerState[fieldNameTraceID] = strconv.FormatInt(sc.raw.TraceID, 10)
-	splitTextCarrier.TracerState[fieldNameSpanID] = strconv.FormatInt(sc.raw.SpanID, 10)
-	splitTextCarrier.TracerState[fieldNameSampled] = strconv.FormatBool(sc.raw.Sampled)
-
-	sc.Lock()
-	if l := len(sc.raw.Baggage); l > 0 && splitTextCarrier.Baggage == nil {
-		splitTextCarrier.Baggage = make(map[string]string, l)
-	}
-	for k, v := range sc.raw.Baggage {
-		splitTextCarrier.Baggage[k] = v
-	}
-	sc.Unlock()
-	return nil
-}
-
-func (p *splitTextPropagator) JoinTrace(
-	operationName string,
-	carrier interface{},
-) (opentracing.Span, error) {
-	splitTextCarrier, ok := carrier.(*opentracing.SplitTextCarrier)
-	if !ok {
-		return nil, opentracing.ErrInvalidCarrier
-	}
-	requiredFieldCount := 0
-	var traceID, propagatedSpanID int64
-	var sampled bool
-	var err error
-	for k, v := range splitTextCarrier.TracerState {
-		switch strings.ToLower(k) {
-		case fieldNameTraceID:
-			traceID, err = strconv.ParseInt(v, 10, 64)
-			if err != nil {
-				return nil, opentracing.ErrTraceCorrupted
-			}
-		case fieldNameSpanID:
-			propagatedSpanID, err = strconv.ParseInt(v, 10, 64)
-			if err != nil {
-				return nil, opentracing.ErrTraceCorrupted
-			}
-		case fieldNameSampled:
-			sampled, err = strconv.ParseBool(v)
-			if err != nil {
-				return nil, opentracing.ErrTraceCorrupted
-			}
-		default:
-			continue
-		}
-		requiredFieldCount++
-	}
-	const expFieldCount = 3
-	if requiredFieldCount < expFieldCount {
-		if len(splitTextCarrier.TracerState) == 0 {
-			return nil, opentracing.ErrTraceNotFound
-		}
-		return nil, opentracing.ErrTraceCorrupted
-	}
-
-	sp := p.tracer.getSpan()
-	sp.raw = RawSpan{
-		Context: Context{
-			TraceID:      traceID,
-			SpanID:       randomID(),
-			ParentSpanID: propagatedSpanID,
-			Sampled:      sampled,
-		},
-	}
-	sp.raw.Baggage = splitTextCarrier.Baggage
 
-	return p.tracer.startSpanInternal(
-		sp,
-		operationName,
-		time.Now(),
-		nil,
-	), nil
+// AccessorCarrier is a flexible carrier interface which can be implemented
+// by types which have a means of storing the trace metadata and already know
+// how to serialize themselves (for example, protocol buffers).
+type AccessorCarrier interface {
+	SetMeta(traceID, spanID int64, sampled bool)
```

Comment:
in other places we refer to this as "Tracer State"... not arguing that one name is better than the other, but consistency would suggest `SetTracerState()`/`TracerState()` (I think Go style discourages the `Get`). Though `GetBaggage` SGTM since it has a non-standard form for an accessor with the closure param, etc.